### ISSUE TITLE
Backup

### ISSUE THREAD

**Issue by killerty69** [GameServerManagers/LinuxGSM#1163]
More of a feature idea than an issue but shouldn't the backup script also back up saves as well, at least put an option to do so as well as just backing up the saves. We could just have a default location that it expects to find them with each game, but still ask where the location is for saves so they can just press enter if it is in the default location or type it out manually.

**Comment by UltimateByte** [comment]
So if i got it, you want the backup to only backup specific files ?

**Comment by dgibbs64** [comment]
I think backing up only configs and saves is a good idea. However the issue we run in to is that different servers do it in different ways. So to be able to add this kind of support is very time consuming and a rather large pain. Not only that what about custom maps and add-ons. For us it would be very problematic to implement.

**Comment by UltimateByte** [comment]
Unless maybe we find a way so that the user can input folders that he wants to save, at its own risk.

**Comment by killerty69** [comment]
I agree with you dgibbs but my idea was just to have them input the location for their saves which we just run backup on the directory. We could have default locations for most games as it won't change, but we don't have to implement a restore function just zip up that folder.

**Comment by UltimateByte** [comment]
I think this is a feature that should be added into a config file put into the lgsmdir. Some work would need to be made to LGSM first to make those kind of things easier, i think.

**Comment by nadal28** [comment]
Agree, in RUST for example I only want to back up one directory, that contains server config files and some .sav files, but when I use ./rustserver backup it started copying 5.6 GB... when the files I only want to backup are only 30mb.

**Comment by UltimateByte** [comment]
So, I got I think, a good idea.

We could allow the user to set one or multiple dirs to backups (default would be known minimal backup) using a config variable, in addition to a new command:

Example variable: 
`backupcustomdirs="${rootdir}/${selfname}; ${systemdir}/dir/subdir; ${systemdir}/dir2; ${systemdir}/server.db; ${lgsmdir}"`

Command: 
`./gameserver backup-custom`

What do you guys think about it?

**Comment by nadal28** [comment]
Yeah its good, at least for me, would be a great addition in my opinion

**Comment by dgibbs64** [comment]
I have been thinking about this. The idea is for me to generate a list of files fro a default install (from steam) with md5 sums. This list will be an exclude list so only files that have been added after the default install will be backed up as it doesn't matter if a default map is deleted ans steamcmd will just re-download it. It may also be possible to use this method as a file downloader for non steam games. Like the new UT that does not have an updater yet. To do an update it would compare all files remotely with the locale ones any that do not match the check sum will be downloaded.

I would have to write a script with a repo of all games to check for updates says once per hour and re generate the list if there are any changes

**Comment by UltimateByte** [comment]
Overkill, but I love it !

**Comment by jimwald1221** [comment]
For what it's worth, I did something similar to what @UltimateByte suggested. I created a variable in the LGSM config pointing to a flat file with lines containing the directories and files I wanted to exclude and changed the tar command in "command_backup.sh" to use -X and point at the file.

It's more on me to update the file but this should require more work on the admin side than on the development side. Any comments?

**Comment by marvinlehmann** [comment]
Kinda relevant too #1563

**Comment by dgibbs64** [comment]
Related to #1595

**Comment by MicLieg** [comment]
link #2106
